tests/functionaltests/Tests/NSURLSession.mm

```diff
@@ -68,6 +69,48 @@ - (NSURLSession*)createSession {
 }
 
 //
+// Property getters
+//
+// Note: Since we're using @synchronized(self), the atomic annotation isn't strong enough for
```

Comment:
@ms-jihua: `@synchronized(self)` locks around self; an atomic property locks only around `&_ivar`. We'd like to guard the mutation of the members _as well as_ their access.